### PR TITLE
Read tensor.db for bestscore

### PR DESCRIPTION
**Pull request by payalcha**
1. Change torch/mnist plan to get tensor.db in local_state.
This change is done in continuation of Eran's creation of tensor.db file. For keras/mnist it was creating in local_state and torch/mnist was getting created in workspace of aggregator.
3. Change summary_helper to get details from tensor.db instead of metrics.
4. Use defused etree for XMLParser.
5. **Documentation** change - Instead of Exceptions write Exception

### CODE REVIEW

**Review by noopurintel — APPROVED** [review]
Suggested a few minor changes offline. Please do those before merging.
Approved.

File: tests/end_to_end/utils/summary_helper.py [hunk 1]
```diff
@@ -32,6 +33,25 @@ def initialize_xml_parser():
...
+        print(f"Database file {database_file} not found. Cannot get best accuracy")
+        return best_accuracy
+
+    db_helper = DBHelper(database_file)
+    round_number, best_score = db_helper.read_key_value_store()
+    print(f"Best accuracy: {best_score} is in round_number {round_number} ")
+    return best_accuracy
+
+
 def get_aggregated_accuracy(agg_log_file):
```
Comment (noopurintel) [inline tests/end_to_end/utils/summary_helper.py]: Please remove this function as it is no longer required.

File: tests/end_to_end/utils/summary_helper.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 
 import tests.end_to_end.utils.constants as constants
 from tests.end_to_end.utils.generate_report import convert_to_json
```
Comment (noopurintel) [inline tests/end_to_end/utils/summary_helper.py]: This import statement can be removed along with get_aggregated_accuracy.

**Review by noopurintel — CHANGES_REQUESTED** [review]

File: tests/end_to_end/utils/summary_helper.py [hunk 3]
```diff
@@ -32,6 +33,25 @@ def initialize_xml_parser():
...
+        database_file: the database file
+    Returns:
+        best_accuracy: the best accuracy
+    """
+    best_accuracy = "Not Found"
+    if not os.path.exists(database_file):
+        print(f"Database file {database_file} not found. Cannot get best accuracy")
+        return best_accuracy
+
+    db_helper = DBHelper(database_file)
```
Comment (noopurintel) [inline tests/end_to_end/utils/summary_helper.py]: Please rename the variable name from `db_helper` to say `db_obj` or `db_helper_obj`, because the file name is also `db_helper.py` and when imported at some place it would be as `db_helper` mostly - so we better avoid this confusion.

**Review by noopurintel — CHANGES_REQUESTED** [review]

File: tests/end_to_end/test_suites/tr_with_fedeval_tests.py [hunk 4]
```diff
@@ -99,9 +100,9 @@ def test_eval_federation_via_dockerized_workspace(request, fx_federation_tr_dws)
...
-    new_metric_file_path = os.path.join(new_fed_obj.aggregator.workspace_path, "logs", "aggregator_metrics.txt")
-    model_accuracy_eval = get_aggregated_accuracy(new_metric_file_path)
-    log.info(f"Model accuracy during evaluation only on prev trained model is : {model_accuracy_eval})")
+    tensor_db_file_path = os.path.join(new_fed_obj.aggregator.workspace_path, "local_state", "tensor.db")
+    best_model_accuracy_eval = get_best_accuracy(tensor_db_file_path)
+    log.info(f"Model accuracy post {request.config.num_rounds} rounds: {best_model_accuracy}")
 
     # verify that the model accuracy is similar to the previous model accuracy max of 1% difference
-    assert abs(model_accuracy - model_accuracy_eval) <= 0.01, "Model accuracy is not similar to the previous model accuracy"
+    assert abs(best_model_accuracy - best_model_accuracy_eval) <= 0.01, "Model accuracy is not similar to the previous model accuracy"
```
Comment (MasterSkepticista) [inline tests/end_to_end/test_suites/tr_with_fedeval_tests.py]: 1% is significant in data science. If logically the expectation is to get same value under floating point error, this tolerance should be around 1e-5.

**Review by MasterSkepticista — CHANGES_REQUESTED** [review]
At all relevant points in the code, replace `accuracy` with `metric` - accuracy is one of thousands of metrics that OpenFL can be used for.

File: tests/end_to_end/utils/summary_helper.py (hunk 2 above)
Comment (payalcha) [reply tests/end_to_end/utils/summary_helper.py]: Done

File: tests/end_to_end/utils/summary_helper.py (hunk 3 above)
Comment (payalcha) [reply tests/end_to_end/utils/summary_helper.py]: Done

File: tests/end_to_end/utils/summary_helper.py (hunk 1 above)
Comment (payalcha) [reply tests/end_to_end/utils/summary_helper.py]: done

File: tests/end_to_end/test_suites/tr_with_fedeval_tests.py (hunk 4 above)
Comment (payalcha) [reply tests/end_to_end/test_suites/tr_with_fedeval_tests.py]: Done. Changed tolerance to 0.00001.

**Review by noopurintel — APPROVED** [review]

**Review by rajithkrishnegowda — APPROVED** [review]
